src/components/code-editor/index.jsx

```diff
@@ -1,99 +1,72 @@
-import { Component, render } from 'preact';
-import codemirror from 'codemirror';
+import { useState, useRef, useEffect } from 'preact/hooks';
+import { EditorView } from 'codemirror';
+import { lineNumbers, keymap, highlightActiveLineGutter, highlightActiveLine } from '@codemirror/view';
+import { EditorState } from '@codemirror/state';
+import { defaultKeymap, history, historyKeymap, indentWithTab } from '@codemirror/commands';
+import { javascript } from '@codemirror/lang-javascript';
+import { syntaxHighlighting, HighlightStyle, indentUnit, bracketMatching } from '@codemirror/language';
+import { tags } from '@lezer/highlight';
+import { closeBrackets, autocompletion } from '@codemirror/autocomplete';
 import cx from '../../lib/cx';
-import 'codemirror/mode/jsx/jsx';
-import 'codemirror/addon/comment/comment';
-import 'codemirror/lib/codemirror.css';
+
 import style from './style.module.css';
 import './code-mirror.css';
 
-export default class CodeEditor extends Component {
-	scratch = document.createElement('div');
-
-	showError(error) {
-		clearTimeout(this.showErrorTimer);
-		if (this.errors) {
-			this.editor.operation(() => {
-				this.errors.forEach(e => e.clear());
-			});
-			this.errors.length = 0;
-		}
-
-		if (!error || !error.loc) return;
+// Custom theme that better matches our Prism config, though
+// the lexer is somewhat limited so it still deviates
```

Comment:
Theme definitely isn't perfect, but this is a bit closer to our Prism config than the previous set up was.